spec/lib/nerve_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'spec_helper'
+require 'nerve/configuration_manager'
+require 'nerve/service_watcher'
+require 'nerve/reporter'
+require 'nerve/reporter/base'
+require 'nerve'
+
+def make_mock_service_watcher
+    mock_service_watcher = double(Nerve::ServiceWatcher)
+    allow(mock_service_watcher).to receive(:start)
+    allow(mock_service_watcher).to receive(:stop)
+    allow(mock_service_watcher).to receive(:alive?).and_return(true)
+    allow(mock_service_watcher).to receive(:was_up).and_return(true)
+    mock_service_watcher
+end
+
+describe Nerve::Nerve do
+  let(:config_manager) { Nerve::ConfigurationManager.new() }
+  let(:mock_config_manager) { double() }
```

Comment:
why do you need both a real and a fake config manager? should this be a real instance double, to make sure you're not invoking or stubbing methods that don't actually exist?